Kudu.Core/Deployment/StructuredTextLogger.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Kudu.Core.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Kudu.Core.Deployment
+{
+    public class StructuredTextLogger : IDetailedLogger
+    {
+        private readonly static object DocumentLock = new object();
```

Comment:
Not quite. Each deployment ends up with a log file. Though during the deployment each time you call `.Log("Step 1")` you get back another instance of the same logger that you can then use to call `.Log("Step 1.1")` which will be a nested message under `"Step 1"`. This is the same way `XmlLogger` works today. Does that make sense?